frontend/views/containers/chatroom/ChatMain.vue

```diff
@@ -283,7 +292,7 @@ export default ({
           prepublish: (message) => {
             const msgValue = JSON.parse(message.opValue())
             const { meta, data } = msgValue
-            this.messages.push({
+            this.messageState.contract.messages.push({
```

Comment:
Regarding issue #1503 and the comment below, you could instead rename this handler from `prepublish` to `postpublish`. Then it would be safe to use `message.hash()` safely as the identifier.

Reply:
Yeah. It's a nice solution in terms of comparing GIMessages.
On the other hand, the pending could work a bit weird, if we add the pending message in `postpublish`.
Because the message won't be added immediately after users press `Enter`.

What do you recommend?
1. Keep it in `prepublish` and update later
2. Move it to `postpublish` and update later

Reply:
I think we should for now put it in `postpublish`, because at least will avoid weird bugs that would be caused by the `messageId` being wrong

Reply:
Okay. Will do.

Reply:
Oh, weird. OK 👍

Reply:
Thanks